src/nft.ts

```diff
@@ -184,4 +186,37 @@ export default class Nft extends AinftBase {
     const trailingUrl = `native/${appId}/${chain}/${network}/${collectionId}/${tokenId}/transfer`;
     return this.sendRequest(HttpMethod.POST, trailingUrl, body);
   }
+
+  uploadAsset({
+    appId,
+    filename,
+    buffer,
+    filePath
+  }: UploadAssetFromBufferParams): Promise<TransactionInput> {
+    const trailingUrl = `asset/${appId}`;
+    return this.sendFormRequest(HttpMethod.POST, trailingUrl, {
+      appId, ...(filePath && {filePath})
+    }, {
+      asset: {
+        filename,
+        buffer
+      }
+    });
+  }
+
+  uploadAssetWithDataUrl({
+    appId,
+    filename,
+    dataUrl,
+    filePath
+  }: UploadAssetFromDataUrlParams): Promise<TransactionInput> {
```

Comment:
if user does not send transaction, image will be deleted?

Reply:
This function does not return txBody, it returns asset url. i will fix it thanks!